### ISSUE TITLE
3.0 release

### ISSUE THREAD

**Issue by jmezach** [rr-wfm/MSBuild.Sdk.SqlProj#630]
I was wondering if we should consider doing a 3.0 release of this SDK, coinciding with the release of .NET 9. Some things I'm considering for this release are:

- [x] Drop support for .NET 6 since it is out-of-support when .NET 9 comes out
- [x] Add support for .NET 9 (I don't think we have that yet)
- [ ] ~~Remove the dependency on the `sqltoolsservice` submodule. We introduced this in https://github.com/rr-wfm/MSBuild.Sdk.SqlProj/issues/23, but it looks like we did that because of https://github.com/microsoft/DacFx/issues/397 which should be fixed now. If we're doing a major update it might be a good time to refactor this.~~
- [x] Create a first official release of #585 targeting .NET Aspire 9

@ErikEJ @jeffrosenberg Thoughts?

**Comment by ErikEJ** [comment]
@jmezach Agree, this makes a lot of sense

**Comment by ErikEJ** [comment]
@jmezach Could we publish a 3.0 preview, I have some users asking for .NET 9 support?

**Comment by ErikEJ** [comment]
@jmezach What are your plans for the Aspire Hosting Integration?

**Comment by jmezach** [comment]
Yeah, I think we can ship a 3.0 preview from `master` right now. Perhaps we should move the Aspire hosting stuff to the CommunityToolkit repo. Not entirely sure what would be required there to make that happen though.

**Comment by ErikEJ** [comment]
> I think we can ship a 3.0 preview from master right now.

Great, thanks!

> Not entirely sure what would be required there to make that happen though.

"Just" submit a PR, there is a guide here: https://github.com/CommunityToolkit/Aspire/blob/main/docs/create-integration.md

**Comment by jmezach** [comment]
Release pipeline for 3.0 preview is running now: https://github.com/rr-wfm/MSBuild.Sdk.SqlProj/actions/runs/11554536334

**Comment by jmezach** [comment]
I'll have a look at creating a PR to the CommunityToolkit repo when I have some time

**Comment by jmezach** [comment]
Shouldn't we have merged #634 first before doing the release?

Also, it looks like the CommunityToolkit repo is still targeting Aspire 8, so I don't think our integration will work with that in its current form.

**Comment by ErikEJ** [comment]
Oh, yes, merge #634 - missed that, sorry.

The community toolkit has an aspire 9 branch that will very soon become main, so no showstopper there, your PR can just target that

**Comment by jmezach** [comment]
Had to open a new issue first before we can get open a PR. https://github.com/CommunityToolkit/Aspire/issues/178.

**Comment by jmezach** [comment]
I think we're pretty much done here. The Aspire prototype has moved to the CommunityToolkit and removing the dependency on the `sqltoolsservice` submodule appears to be a much bigger effort than I anticipated. Let's close this and get the 3.0 release out the door.

**Comment by ErikEJ** [comment]
@jmezach Looks like 3.0.0-beta.3 was published rather than 3.0.0

**Comment by jmezach** [comment]
Yes, we had some changes on the CI pipeline so I wanted to make sure that everything still works. I'll bump the version number now.

**Comment by ErikEJ** [comment]
@jmezach There seems to be some flakiness (with the new SDK??) - I got the build hhorugh, and created some release notes here: https://github.com/rr-wfm/MSBuild.Sdk.SqlProj/releases/tag/v3.0.0

**Comment by jmezach** [comment]
Yeah, not entirely sure what's going wrong there. Haven't seen this behaviour before.

**Comment by ErikEJ** [comment]
I think they have made tests more parallel

https://learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-9/sdk#unit-testing

Mabe we should add `TestTfmsInParallel` false to the test project?
